### PR TITLE
NOISSUE - Update docs - services heartbeat 

### PR DESCRIPTION
**Pull request by mteodor**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,7 +103,7 @@ Environment:
 |	MF_AGENT_MQTT_CLIENT_PK                | Location of client certificate key for MTLS                   | thing.key                         |
 
 Here `thing` is a Mainflux thing, and control channel from `channels` is used with `req` and `res` subtopic
-(i.e. app needs to PUB/SUB on `/channels/<channel_id>/messages/req` and `/channels/<channel_id>/messages/res`).
+(i.e. app needs to PUB/SUB on `/channels/<control_channel>/messages/req` and `/channels/<control_channel>/messages/res`).
```
Comment (manuio) [inline README.md:106]: `control_channel_id` ?

Comment (mteodor) [reply README.md:106]: @drasko `control_channel` or `control_channel_id`?

Comment (drasko) [reply README.md:106]: I can accept this, it is clear enough and short/simple enough.

**Comment by drasko** [comment]
@mteodor btw - why do we have a definition of the `File` variable in this TOML: https://github.com/mainflux/agent#config

This looks redundant to me, please remove it.

File: README.md (hunk 1 above)
Comment (manuio) [reply README.md:106]: @drasko If it's an ID it's not clear. It could be a name.

Comment (drasko) [reply README.md:106]: I agree, adding `_id` does not cost much and it is more precise.

File: README.md [hunk 2]
```diff
@@ -88,7 +86,7 @@ Environment:
 |	MF_AGENT_BOOTSTRAP_KEY                 | Mainflux boostrap key                                         |                                   |
 |	MF_AGENT_BOOTSTRAP_RETRIES             | Number of retries for bootstrap procedure                     | 5                                 |
 |	MF_AGENT_BOOTSTRAP_RETRY_DELAY_SECONDS | Number of seconds between retries                             | 10                                |
-|	MF_AGENT_CONTROL_CHANNEL               | Channel for sending controls, commands                        |                                   |
+|	MF_AGENT_control_channel_id               | Channel for sending controls, commands                        |                                   |
```
Comment (manuio) [inline README.md:89]: Capital letters here

File: README.md [hunk 3]
```diff
@@ -103,22 +101,73 @@ Environment:
...
+mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_channel_id>/messages/req -h <mqtt_host> -p 1883  -m  '[{"bn":"1:", "n":"config", "vs":"view"}]'  
+```
+Check the output in terminal where you subscribed for results. You should see something like:
+```json
+[
+  {
+    "bn": "1",
+    "n": "view",
+    "t": 1588091188.8872917,
+    "vs": "[{\"Name\":\"duster\",\"LastSeen\":\"2020-04-28T18:06:56.158130519+02:00\",\"Status\":\"offline\",\"Type\":\"test\",\"Terminal\":0},{\"Name\":\"scrape\",\"LastSeen\":\"2020-04-28T18:06:39.58849766+02:00\",\"Status\":\"offline\",\"Type\":\"test\",\"Terminal\":0}]"
```
Comment (manuio) [inline README.md:167]: Add some blank lines in this section

Comment (mteodor) [reply README.md:167]: @manuio have you previewed [README](https://github.com/mainflux/agent/blob/71029eb7ba93b5c67480f03c0bceec86c07c03fd/README.md), I think more blanks would be to much

Comment (manuio) [reply README.md:167]: It should change nothing in terms of visualization

File: README.md [hunk 4]
```diff
@@ -102,23 +105,97 @@ Environment:
...
+    "Status": "offline",
+    "Type": "test",
+    "Terminal": 0
+  },
+  {
+    "Name": "scrape",
+    "LastSeen": "2020-04-28T18:06:39.58849766+02:00",
+    "Status": "offline",
+    "Type": "test",
+    "Terminal": 0
```
Comment (manuio) [inline README.md:151]: Maybe snake_case here?

File: README.md [hunk 5]
```diff
@@ -102,23 +105,97 @@ Environment:
...
+    "t": 1588091188.8872917,
+    "vs": "[{\"name\":\"duster\",\"last_seen\":\"2020-04-28T18:06:56.158130519+02:00\",\"status\":\"offline\",\"type\":\"test\",\"terminal\":0},{\"name\":\"scrape\",\"last_seen\":\"2020-04-28T18:06:39.58849766+02:00\",\"status\":\"offline\",\"type\":\"test\",\"terminal\":0}]"
+  }
+]
+```
+
+
+## How to save config via agent
+Agent can be used to send configuration file for the [Export][export] service from cloud to gateway via MQTT.
+Here is the example command
```
Comment (drasko) [inline README.md:186]: `Here is the example command:` (missing colon)

File: README.md [hunk 6]
```diff
@@ -102,23 +105,97 @@ Environment:
...
 
 Example of on command can be:
 
+```bash
+mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_channel_id>/messages/services/adc -h <mqtt_host> -p 1883  -m  "[{\"bn\":\"1:\", \"n\":\"read\", \"vs\":\"temperature\"}]"
+```
+
+## Heartbeat service
+Services running on the same host can publish to `heartbeat.<service-name>.<service-type>` a heartbeat message.  
+Agent will keep a record on those service and update their `live` status. If heartbeat is not received in 10 sec it marks it `offline`.
```
Comment (drasko) [inline README.md:130]: Break long lines in multiple (markdown formatter wouldnot mind).

File: README.md [hunk 7]
```diff
@@ -102,23 +105,97 @@ Environment:
...
+Upon next heartbeat service will be marked `online` again.
+To check services that are currently registered to agent you can:
```
Comment (drasko) [inline README.md:132]: Whiteline before this sentence

File: README.md [hunk 8]
```diff
@@ -102,23 +105,99 @@ Environment:
...
+]
+```
+
+
+## How to save config via agent
+Agent can be used to send configuration file for the [Export][export] service from cloud to gateway via MQTT.
+Here is the example command:
+
+```bash
+mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_ch_id>/messages/req -h localhost -p 18831  -m  "[{\"bn\":\"1:\", \"n\":\"config\", \"vs\":\"config.toml, RmlsZSA9ICIuLi9jb25maWdzL2NvbmZpZy50b21sIgoKW2V4cF0KICBsb2dfbGV2ZWwgPSAiZGVidWciCiAgbmF0cyA9ICJuYXRzOi8vMTI3LjAuMC4xOjQyMjIiCiAgcG9ydCA9ICI4MTcwIgoKW21xdHRdCiAgY2FfcGF0aCA9ICJjYS5jcnQiCiAgY2VydF9wYXRoID0gInRoaW5nLmNydCIKICBjaGFubmVsID0gIiIKICBob3N0ID0gInRjcDovL2xvY2FsaG9zdDoxODgzIgogIG10bHMgPSBmYWxzZQogIHBhc3N3b3JkID0gImFjNmI1N2UwLTliNzAtNDVkNi05NGM4LWU2N2FjOTA4NjE2NSIKICBwcml2X2tleV9wYXRoID0gInRoaW5nLmtleSIKICBxb3MgPSAwCiAgcmV0YWluID0gZmFsc2UKICBza2lwX3Rsc192ZXIgPSBmYWxzZQogIHVzZXJuYW1lID0gIjRhNDM3ZjQ2LWRhN2ItNDQ2OS05NmI3LWJlNzU0YjVlOGQzNiIKCltbcm91dGVzXV0KICBtcXR0X3RvcGljID0gIjRjNjZhNzg1LTE5MDAtNDg0NC04Y2FhLTU2ZmI4Y2ZkNjFlYiIKICBuYXRzX3RvcGljID0gIioiCg==\"}]"
```
Comment (drasko) [inline README.md:191]: We do not fullblown cert string here, replace by `<cert>`

Comment (drasko) [reply README.md:191]: Explain also which kind of cert this is - CA, pub or priv X.509. For example `<ca_cert>` or `<pub_cert_x509>` or something...

File: README.md [hunk 9]
```diff
@@ -102,23 +105,104 @@ Environment:
...
+]
+```
+
+
+## How to save config via agent
+Agent can be used to send configuration file for the [Export][export] service from cloud to gateway via MQTT.  
+Here is the example command:
+
+```bash
+mosquitto_pub -u <thing_id> -P <thing_key> -t channels/<control_channel_id>/messages/req -h localhost -p 1883  -m  "[{\"bn\":\"1:\", \"n\":\"config\", \"vs\":\"<config_file_path>, <file_cont_base64>\"}]"
```
Comment (drasko) [inline README.md:191]: Replace `<file_cont_base64>` by `<file_content_base64>`. It hineders readility (`cont` is usually short for `continue`), 3 letters will not change things here.

**Pull request merged**